### ISSUE TITLE
Merge twisted.trial._utilpy3 back into twisted.trial.util

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6019]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6019|
|Type|enhancement|
|Created|2012-09-24 18:11:30Z|
|Branch|https://github.com/twisted/twisted/tree/merge-trial-util-utilpy3-6019|

`twisted.trial._utilpy3` is no longer necessary, and can be merged back into trial (once [#6009](https://github.com/twisted/twisted/issues/6009) is merged).

<details><summary>Searchable metadata</summary>

```
trac-id__6019 6019
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_merge_trial_util_utilpy3_6019 branches/merge-trial-util-utilpy3-6019
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1348510290000000 1348510290000000
changetime__1349283673000000 1349283673000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [35879]) Branching to 'merge-trial-util-utilpy3-6019'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [35880]) Move utilpy3 code into util; move test_utilpy3 code into test_util; subclass SynchronousTestCase instead of stdlib TestCase.

refs [#6019](https://github.com/twisted/twisted/issues/6019)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [35882]) Update random code that was using twisted.trial._utilpy3

refs [#6019](https://github.com/twisted/twisted/issues/6019)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @itamarst<br>@exarkun set status to `new`|
|-|-|

Okay: http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/merge-trial-util-utilpy3-6019

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_6"></a>|@itamarst set owner to @exarkun|
|-|-|

Looks good, merge after one fix: you forgot to remove `twisted.trial.test.test_utilpy3` from `admin/_twistedpython3.py`.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [35909]) Merge merge-trial-util-utilpy3-6019 

Author: exarkun
Reviewer: itamar
Fixes: [#6019](https://github.com/twisted/twisted/issues/6019)

Merge the temporary porting aid `twisted.trial._utilpy3` back into `twisted.trial.util` now that
all of `twisted.trial.util` has been ported to Python 3.
